src/ipc/ipc4/dai.c

```diff
@@ -83,6 +84,30 @@ int dai_config_dma_channel(struct dai_data *dd, struct comp_dev *dev, const void
 #endif
 		break;
 	case SOF_DAI_INTEL_HDA:
+#if defined(CONFIG_ACE_VERSION_2_0)
+		if (copier_cfg->gtw_cfg.node_id.f.dma_type == ipc4_alh_link_output_class ||
+		    copier_cfg->gtw_cfg.node_id.f.dma_type == ipc4_alh_link_input_class) {
+			struct processing_module *mod = comp_get_drvdata(dev);
+			struct copier_data *cd = module_get_private_data(mod);
+
+			channel = DMA_CHAN_INVALID;
+
+			if (!cd->gtw_cfg) {
+				comp_err(dev, "No gateway config found!");
+				return channel;
```

Comment:
I think it would be more obvious to have `return DMA_CHAN_INVALID` here and move line 93 to around line 100 below